drivers/storage/ide/pciidex/fdo.c

```diff
@@ -4,488 +4,529 @@
  * FILE:            drivers/storage/pciidex/fdo.c
  * PURPOSE:         IRP_MJ_PNP operations for FDOs
  * PROGRAMMERS:     Herv� Poussineau (hpoussin@reactos.org)
+ *                  Dmitry Borisov (di.sean@protonmail.com)
  */
 
 #include "pciidex.h"
 
 #define NDEBUG
 #include <debug.h>
 
-#include <initguid.h>
-#include <wdmguid.h>
+static
+CODE_SEG("PAGE")
+NTSTATUS
+PciIdeXFdoParseResources(
+    _In_ PFDO_DEVICE_EXTENSION FdoExtension,
+    _In_ PCM_RESOURCE_LIST ResourcesTranslated)
+{
+    PCM_PARTIAL_RESOURCE_DESCRIPTOR BusMasterDescriptor = NULL;
+    PCM_PARTIAL_RESOURCE_DESCRIPTOR InterruptDescriptor = NULL;
+    ULONG i;
+    PVOID IoBase;
+
+    PAGED_CODE();
+
+    if (!ResourcesTranslated)
+        return STATUS_INSUFFICIENT_RESOURCES;
```

Comment:
`PciIdeXFdoParseResources` doesn't seem to allocate chunk of memory for `ResourcesTranslated` but rather it expects such a parameter to be valid. Perhaps you could just return `STATUS_INVALID_PARAMETER` here or whatever?